charts/gitprime-app/v2.0.0/templates/dataPipeline/_deployment.tpl

```diff
@@ -0,0 +1,114 @@
+{{- /* This represents a deployment that runs the main Data Pipeline jar and moves into different modes based on */}}
+{{- /* environment variables and the like. */}}
+{{- define "dataPipeline.deploymentTemplate" }}
+{{- $globalEnvironment := .Values.dataPipeline.environment }}
+{{- $environment := .templateData.environment }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: gitprime-dp-{{- template "helpers.environment.fullName" .}}-{{ .templateData.operationMode }}
+  labels:
+    app: gitprime-dp-{{- template "helpers.environment.fullName" .}}-{{ .templateData.operationMode }}
+spec:
+  replicas: {{ .templateData.replicaCount }}
+  selector:
+    matchLabels:
+      app: gitprime-dp-{{- template "helpers.environment.fullName" .}}-{{ .templateData.operationMode }}
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: gitprime-dp-{{- template "helpers.environment.fullName" .}}-{{ .templateData.operationMode }}
+    spec:
+      containers:
+        - name: gitprime-dp-{{- template "helpers.environment.fullName" .}}-{{ .templateData.operationMode }}
+          image: gp-docker.gitprime-ops.com/integrations/gitprime-data-pipeline:{{ .Values.dataPipeline.build.commitSHA }}
+          imagePullPolicy: IfNotPresent
+          env:
+            - name: SYSTEM_ENV_PARENT
+              value: {{ quote .Values.environment.parentName }}
+            - name: SYSTEM_ENV_MODIFIER
+              value: {{ quote .Values.environment.modifierValue }}
+            - name: SYSTEM_ENV
+              value: {{ include "helpers.environment.fullName" . | quote }}
+            - name: SYSTEM_POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: SYSTEM_OPERATION_MODE
+              value: {{ quote .templateData.operationMode }}
+            - name: SYSTEM_CONFIG_SERVER_URL
+              value: {{ include "helpers.configServerURL" . }}
+            - name: GP_REQUEUE_AOD
+              value: {{ quote .templateData.requeueAOD | default true }}
+            - name: GP_LOG_LEVEL
+              value: {{ quote .templateData.logLevel | default "INFO" | lower }}
+          {{- if .templateData.javaOptions }}
+            - name: JAVA_OPTS
+              value: {{ quote .templateData.javaOptions }}
+          {{- end }}
+          {{- if .templateData.maxConcurrentJobs }}
+            - name: GP_MAX_CONCURRENT_JOBS
+              value: {{ quote .templateData.maxConcurrentJobs }}
```

Comment:
The config.json will now have separate values for each queue type, each value.  So things like `GP_MAX_CONCURRENT_JOBS` need to be given with the queue type.

In practice, we will only run one queue type per pod in production.  However, we want to be able to run multiple queues in one pod in other environments, which means we will need to separate out each of these per-queue configs.